core/src/main/c/share/bit_vector.h

```diff
@@ -0,0 +1,75 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2023 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+#ifndef QUESTDB_BIT_VECTOR_H
+#define QUESTDB_BIT_VECTOR_H
+#include "../share/simd.h"
+
+// Custom implementation of std::vector<bool>
+// which is a problem to link on Windows, and in general std::vector<bool> may be dropped from STL
+template<typename T>
```

Comment:
can be simplified to :
```
template<typename T>
class bit_vector_t {
public:
    bit_vector_t() {
        impl = nullptr;
    }

    ~bit_vector_t() {
        impl = static_cast<T *>(realloc(impl, 0));
    }

    void reset(size_t length) {
        const auto byte_len = sizeof(T) * (length / sizeof(T) + 1);
        impl = static_cast<T *>(realloc(impl, byte_len));
        memset(impl, 0, byte_len);
    }

    inline void set(size_t index) {
        const auto pos = index / sizeof(T);
        const auto bit = index & (sizeof(T) - 1);
        impl[pos] |= (1 << bit);
    }

    inline bool operator[](size_t index) const {
        const auto pos = index / sizeof(T);
        const auto bit = index & (sizeof(T) - 1);
        return impl[pos] & (1 << bit);
    }

private:
    T *impl;
};
```

Reply:
Thanks for looking. I found a few bugs in the old code and had to re-write it, please have another look when you have a moment.